### PR TITLE
MOB-9309 Support nested field types

### PR DESCRIPTION
**Pull request by megha-iterable**
## Jira Ticket(s)

* [MOB-9309](https://iterable.atlassian.net/browse/MOB-9309)

[MOB-9309]: https://iterable.atlassian.net/browse/MOB-9309?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: tests/unit-tests/NestedFieldSupportForArrayData.swift [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+                            "lengthInches": 20,
+                            "widthInches": 30
+                        ],
+                    ]
+                ]
+            ]
+        ]
+
+        let matchedCriteriaId = CriteriaCompletionChecker(anonymousCriteria: data(from: mockData)!, anonymousEvents: eventItems).getMatchedCriteria()
+        XCTAssertEqual(matchedCriteriaId, nil)
```
Comment (evantk91) [inline tests/unit-tests/NestedFieldSupportForArrayData.swift:132]: Same comment here as I left for Android.

**Review by evantk91 — COMMENTED** [review]
Same comment as I left for Android.

File: tests/unit-tests/NestedFieldSupportForArrayData.swift (hunk 1 above)
Comment (megha-iterable) [reply tests/unit-tests/NestedFieldSupportForArrayData.swift:132]: Updated the logic as per the PR comment

**Review by evantk91 — APPROVED** [review]
LGTM

**Review by evantk91 — APPROVED** [review]
LGTM

**Pull request merged**